.github/workflows/lint.yml

```
@@ -11,18 +11,17 @@ jobs:
     steps:
       - uses: actions/checkout@v5
 
-      - name: Install dependency manager
-        run: pipx install poetry
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
 
       - name: Set up Python 3.x
         id: setup-python
         uses: actions/setup-python@v6
         with:
           python-version-file: pyproject.toml
-          cache: 'poetry'
 
       - name: Install packages
-        run: poetry install --no-interaction
+        run: uv sync --all-groups
```

I recommend:

```suggestion
        run: uv sync --locked --all-groups
```

https://docs.astral.sh/uv/concepts/projects/sync/